lib/tower_cli/api.py

```diff
@@ -88,16 +88,34 @@ def request(self, method, url, *args, **kwargs):
         if headers.get('Content-Type', '') == 'application/json':
             kwargs['data'] = json.dumps(kwargs.get('data', {}))
 
-        # Disable urllib3 warnings # still warn if verrify_ssl None
+        # Decide whether to require SSL verification
+        verify_ssl = True
         if (settings.verify_ssl is False) or hasattr(settings, 'insecure'):
-            urllib3.disable_warnings()
+            verify_ssl = False
 
         # Call the superclass method.
         try:
             with warnings.catch_warnings():
-                r = super(Client, self).request(method, url, *args,
-                                                verify=False, **kwargs)
+                warnings.simplefilter(
+                    "ignore", urllib3.exceptions.InsecureRequestWarning)
```

Comment:
The "with" block and with the simplefilter function are both necessary in order to suppress the warning. If the user set verify_ssl=false, then they have demonstrated consent to any risks, so they shouldn't want to see additional warnings.